src/controllers/sold.land.filter.controller.ts

```diff
@@ -28,18 +33,34 @@ export const get = (req: Request, res: Response, next: NextFunction) => {
   }
 };
 
-export const post = (req: Request, res: Response, next: NextFunction) => {
+export const post = async (req: Request, res: Response, next: NextFunction) => {
   try {
+
     logger.debugRequest(req, `POST ${config.SOLD_LAND_FILTER_PAGE}`);
-    const hasSoldLand = req.body[HasSoldLandKey];
 
-    setExtraData(req.session, { ...getApplicationData(req.session), [HasSoldLandKey]: hasSoldLand });
+    const session = req.session as Session;
+    const hasSoldLand = (req.body[HasSoldLandKey]).toString();
+    const appData: ApplicationData = getApplicationData(session);
+    appData[HasSoldLandKey] = hasSoldLand;
+
+    let nextPageUrl: string = "";
 
-    if (hasSoldLand === '1') {
-      return res.redirect(config.CANNOT_USE_URL);
-    } else if (hasSoldLand === '0') {
-      return res.redirect(config.SECURE_REGISTER_FILTER_URL);
+    if (hasSoldLand === "1") {
```

Comment:
It might be better to use a guard clause approach to avoid having nested if else.
This could improve readability.